src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -74,15 +75,12 @@ def write_package_info(self, package_name, package_ver, package_size, install_du
                 self.write_event(message, Constants.TelemetryEventLevel.Informational)
 
     # Composed payload
-    def set_machine_config_info(self):
+    def set_and_write_machine_config_info(self):
         # Machine info - sent only once at the start of the run
-        self.machine_info = {
-            'platform_name': str(self.env_layer.platform.linux_distribution()[0]),
-            'platform_version': str(self.env_layer.platform.linux_distribution()[1]),
-            'machine_cpu': self.get_machine_processor(),
-            'machine_arch': str(self.env_layer.platform.machine()),
-            'disk_type': self.get_disk_type()
-        }
+        self.machine_info = "[platform_name={0}][platform_version={1}][machine_cpu={2}][machine_arch={3}][disk_type={4}]".format(
```

Comment:
Not for this PR but we should rationalize this for repeated logging later. I'd suggest not taking a hard dependency on these fields until discussed and settled.

Reply:
You can resolve after it's read.

Reply:
Sure, resolving this comment